commons/protocol/src/main/scala/sbt/protocol/Keys.scala

```diff
@@ -98,3 +98,37 @@ object MinimalBuildStructure {
   implicit val unpickler: Unpickler[MinimalBuildStructure] = genUnpickler[MinimalBuildStructure]
   implicit val pickler: Pickler[MinimalBuildStructure] = genPickler[MinimalBuildStructure]
 }
+
+final case class Attributed[T](data: T)
+object Attributed {
+
+  implicit def unpickler[T: Unpickler]: Unpickler[Attributed[T]] = new Unpickler[Attributed[T]] {
+
+    override val tag = implicitly[FastTypeTag[Attributed[T]]]
+
+    override def unpickle(tag: String, reader: PReader): Any = {
+      reader.hintTag(pickling.FastTypeTag.String)
```

Comment:
You always need to hint.  You shuld just use your own tag here though, instead of "string", so `reader.hintTag(this.tag)`